libraries/botbuilder-dialogs-adaptive/src/conditions/onCondition.ts

```diff
@@ -5,21 +5,121 @@
  * Copyright (c) Microsoft Corporation. All rights reserved.
  * Licensed under the MIT License.
  */
-import { Dialog, DialogEvent } from 'botbuilder-dialogs';
-import { SequenceContext, ActionChangeList } from '../sequenceContext';
-
-export interface OnCondition {
-    /**
-     * Array of actions that should be registered with the `PlanningDialog` when the rule is added.
-     */
-    readonly actions: Dialog[];
+import { Dialog } from 'botbuilder-dialogs';
+import { Expression, ExpressionParserInterface, ExpressionType, Constant, ExpressionEngine } from 'botframework-expressions';
+import { SequenceContext, ActionChangeList, ActionState, ActionChangeType } from '../sequenceContext';
 
+export class OnCondition {
     /**
      * Evaluates the rule and returns a predicted set of changes that should be applied to the
      * current plan.
      * @param planning Planning context object for the current conversation.
      * @param event The current event being evaluated.
      * @param preBubble If `true`, the leading edge of the event is being evaluated.
      */
-    evaluate(planning: SequenceContext, event: DialogEvent, preBubble: boolean): Promise<ActionChangeList[]|undefined>;
+    // evaluate(planning: SequenceContext, event: DialogEvent, preBubble: boolean): Promise<ActionChangeList[] | undefined>;
+
+    private _extraConstraints: Expression[] = [];
+    private _fullConstraint: Expression = null;
+
+
+    /**
+     * Gets or sets the condition which needs to be met for the actions to be executed (OPTIONAL).
+     */
+    public condition: string;
+
+    /**
+     * Gets or sets the actions to add to the plan when the rule constraints are met.
+     */
+    public actions: Dialog[] = [];
+
+    /**
+     * Create a new `OnCondition` instance.
+     * @param condition (Optional) The condition which needs to be met for the actions to be executed.
+     * @param actions (Optional) The actions to add to the plan when the rule constraints are met.
+     */
+    constructor(condition: string = null, actions: Dialog[] = []) {
```

Comment:
>null [](start = 36, length = 4)

null is evil in JS and we've tried to avoid its use in the SDK.  use undefined instead or you can just say condition?: string,